submariner-operator/issue-149-node-placement.md

```diff
@@ -0,0 +1,87 @@
+# Node Placement
+
+[Add support for nodeSelector and toleration scheduling for submariner pods](https://github.com/submariner-io/enhancements/issues/149)
+
+## Summary
+
+Submariner deploys many different agents as part of its core deployment. Gateway, Globalnet Agent and
+Metrics Proxy are deployed on designated gateway nodes, while Route Aents are deployed on all nodes. There is no way
+for users to control or limit where the rest of the agent pods are deployed. Some users want to run these agents on
+specific nodes for billing or networking reasons. Users can't add tolerations directly to these agents because they'll be
+overwritten by the operator. So, users require an easily configurable way to specify these parameters.
+
+## Proposal
+
+This enhancement proposes adding a new field to `Submariner` and `ServiceDiscovery` for users to configure node
+scheduling. This configuration will then be applied to pods of following submariner agents:
+
+* submariner-operator
+* submariner-lighthouse-agent
+* submariner-lighthouse-coredns
+* submariner-networkplugin-syncer
+
+Following agents will be excluded because these are run on Gateway nodes. Users already control which nodes get labeled
+as gateways and ideally we don't want to run anything else on these nodes.
```

Comment:
`ideally we don't want to run anything else on these nodes.`

This line to me means that the 4 other agents (submariner-operator, submariner-lighthouse-agent, submariner-lighthouse-coredns, submariner-networkplugin-syncer) won't be running on gateway nodes. Is that true?

Reply:
This mainly refers to user's services, but we should prefer not to run anything else on GW nodes to keep CPU free to tunnel data traffic. But that discussion is outside the scope of this PR. Scope of this is mainly to differentiate submariner pods from user pods.